lib/DDG/Rewrite.pm

```diff
@@ -137,10 +142,20 @@ sub _build_nginx_conf {
 	my $wrap_string_callback = $self->has_callback && $self->wrap_string_callback;
 	my $uses_echo_module = $wrap_jsonp_callback || $wrap_string_callback;
 	my $callback = $self->callback;
+	my ($spice_name) = $self->path =~ m{^/js/spice/(.+)/$};
+	$spice_name =~ s|/|_|og if $spice_name;
 
 	my $cfg = "location ^~ ".$self->path." {\n";
 	$cfg .= "\tproxy_set_header Accept '".$self->accept_header."';\n" if $self->accept_header;
-	
+
+	if ( $self->has_post_body ) {
+		$cfg .= "\tproxy_method POST;\n";
+		$cfg .= "\tproxy_set_body '" . $self->post_body . "';\n";
+		my $cache_keys = join '', sort keys {
+			map { $_ => 1 } ( $self->post_body =~ m/\$[0-9]+/g )
+		};
+		$cfg .= "\tproxy_cache_key spice_${spice_name}_$cache_keys;\n"
```

Comment:
I'm not sure how universally applicable this is, e.g. if you look at the [random.org API](https://api.random.org/json-rpc/1/basic), the POST body would not change between requests.

You would not cache these particular responses anyway... the question is, will an API response you wish to cache always have some sort of capture variable?

If so, should we construct our own key or embrace a different sort of potential brokenness by [adding our own dummy parameters to the target URI](https://github.com/duckduckgo/zeroclickinfo-spice/pull/2794/commits/8788186649b7840f2c63d126df96e71910409576#diff-903139619c5465ed682525017ef1073eR8).

I think the former is probably what we want - requires less developer intervention.